activation/poet.go

```diff
@@ -338,25 +351,35 @@ func newPoetClient(db poetDbAPI, server types.PoetServer, cfg PoetConfig, logger
 		proofMembers:   make(map[string][]types.Hash32, 1),
 	}
 
+	for _, opt := range opts {
+		opt(poetClient)
+	}
+
 	return poetClient, nil
 }
 
-func (c *PoetClient) Address() string {
+func (c *poetClient) Address() *url.URL {
 	return c.client.Address()
 }
 
-func (c *PoetClient) Submit(
+func (c *poetClient) authorize(
 	ctx context.Context,
-	deadline time.Time,
-	prefix, challenge []byte,
-	signature types.EdSignature,
 	nodeID types.NodeID,
-) (*types.PoetRound, error) {
-	logger := c.logger.With(
-		log.ZContext(ctx),
-		zap.String("poet", c.Address()),
-		log.ZShortStringer("smesherID", nodeID),
-	)
+	challenge []byte,
+	logger *zap.Logger,
```

Comment:
Why is this logger passed as argument? `poetClient` has a `logger` field?

Reply:
Because it gets a logger already enriched with additional fields like context and smesher ID.

Reply:
Not part of this PR, but it doesn't make sense how we handle loggers and contexts to me at the moment. Ideally we would have a situation like this:

```go
ctx = log.WithFields(ctx, log.ZShortStringer("smesherID", nodeID) // probably already in node.go and passed to component
// ... some code to determine which poet to use
ctx = log.WithFields(ctx, zap.String("poet", client.Address())
// ... only pass ctx to child functions
// within child function
logger := c.log.WithOptions(log.ZContext(ctx))
logger.Info("some message") // prints message + smesherID field + poet address field
```

Right now we are passing both the logger and the context around and store information that should be added to logs in both the context and local loggers...